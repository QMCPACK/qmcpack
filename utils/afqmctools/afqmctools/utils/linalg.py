import numpy
import time
from pyscf import lib
from afqmctools.utils.io import (
        format_fixed_width_floats,
        format_fixed_width_strings,
        )


def get_ortho_ao(cell, kpts, LINDEP_CUTOFF=0):
    """Generate canonical orthogonalization transformation matrix.

    Parameters
    ----------
    cell : :class:`pyscf.pbc.cell' object.
        PBC cell object.
    kpts : :class:`numpy.array'
        List of kpoints.
    LINDEP_CUTOFF : float
        Linear dependency cutoff. Basis functions whose eigenvalues lie below
        this value are removed from the basis set. Should be set in accordance
        with value in pyscf (pyscf.scf.addons.remove_linear_dep_).

    Returns
    -------
    X : :class:`numpy.array`
        Transformation matrix.
    nmo_per_kpt : :class:`numpy.array`
        Number of OAOs orbitals per kpoint.
    """
    kpts = numpy.reshape(kpts,(-1,3))
    nkpts = len(kpts)
    nao = cell.nao_nr()
    s1e = lib.asarray(cell.pbc_intor('cint1e_ovlp_sph',hermi=1,kpts=kpts))
    X = numpy.zeros((nkpts,nao,nao),dtype=numpy.complex128)
    nmo_per_kpt = numpy.zeros(nkpts,dtype=numpy.int32)
    for k in range(nkpts):
        sdiag, Us = numpy.linalg.eigh(s1e[k])
        nmo_per_kpt[k] = sdiag[sdiag>LINDEP_CUTOFF].size
        norm = numpy.sqrt(sdiag[sdiag>LINDEP_CUTOFF])
        X[k,:,0:nmo_per_kpt[k]] = Us[:,sdiag>LINDEP_CUTOFF] / norm
    return X, nmo_per_kpt

def get_ortho_ao_mol(S, LINDEP_CUTOFF=0):
    """Generate canonical orthogonalization transformation matrix.

    Parameters
    ----------
    S : :class:`numpy.ndarray`
        Overlap matrix.
    LINDEP_CUTOFF : float
        Linear dependency cutoff. Basis functions whose eigenvalues lie below
        this value are removed from the basis set. Should be set in accordance
        with value in pyscf (pyscf.scf.addons.remove_linear_dep_).

    Returns
    -------
    X : :class:`numpy.array`
        Transformation matrix.
    """
    sdiag, Us = numpy.linalg.eigh(S)
    X = Us[:,sdiag>LINDEP_CUTOFF] / numpy.sqrt(sdiag[sdiag>LINDEP_CUTOFF])
    return X

def modified_cholesky_direct(M, tol=1e-5, verbose=False, cmax=20):
    """Modified cholesky decomposition of matrix.

    See, e.g. [Motta17]_

    Parameters
    ----------
    M : :class:`numpy.ndarray`
        Positive semi-definite, symmetric matrix.
    tol : float
        Accuracy desired. Optional. Default : False.
    verbose : bool
        If true print out convergence progress. Optional. Default : False.
    cmax : int
        Number of cholesky vectors to store N_gamma = cmax M.

    Returns
    -------
    chol_vecs : :class:`numpy.ndarray`
        Matrix of cholesky vectors.
    """
    # matrix of residuals.
    delta = numpy.copy(M.diagonal())
    nchol_max = int(cmax*M.shape[0]**0.5)
    # index of largest diagonal element of residual matrix.
    nu = numpy.argmax(numpy.abs(delta))
    delta_max = delta[nu]
    if verbose:
        print ("# max number of cholesky vectors = %d"%nchol_max)
        header = ['iteration', 'max_residual', 'time']
        print(format_fixed_width_strings(header))
        init = [delta_max]
        print('{:17d} '.format(0)+format_fixed_width_floats(init))
        # print ("# iteration %d: delta_max = %f"%(0, delta_max.real))
    # Store for current approximation to input matrix.
    Mapprox = numpy.zeros(M.shape[0], dtype=M.dtype)
    chol_vecs = numpy.zeros((nchol_max, M.shape[0]), dtype=M.dtype)
    nchol = 0
    chol_vecs[0] = numpy.copy(M[:,nu])/delta_max**0.5
    while abs(delta_max) > tol:
        # Update cholesky vector
        start = time.time()
        Mapprox += chol_vecs[nchol]*chol_vecs[nchol].conj()
        delta = M.diagonal() - Mapprox
        nu = numpy.argmax(numpy.abs(delta))
        delta_max = numpy.abs(delta[nu])
        nchol += 1
        Munu0 = numpy.dot(chol_vecs[:nchol,nu].conj(), chol_vecs[:nchol,:])
        chol_vecs[nchol] = (M[:,nu] - Munu0) / (delta_max)**0.5
        if verbose:
            step_time = time.time() - start
            out = [delta_max, step_time]
            print('{:17d} '.format(nchol)+format_fixed_width_floats(out))

    return numpy.array(chol_vecs[:nchol])
