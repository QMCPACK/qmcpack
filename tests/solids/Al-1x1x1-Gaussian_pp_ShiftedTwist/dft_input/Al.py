#!/usr/bin/env python
  
'''
Gamma point post-HF calculation needs only real integrals.
Methods implemented in finite-size system can be directly used here without
any modification.
'''


import numpy as np
from pyscf import lib
from pyscf.pbc import gto, scf, dft
from pyscf import gto as Mgto
from pyscf.pbc import df
from pyscf.pbc import ao2mo
from pyscf.pbc import tools
from pyscf.pbc.tools.pbc import super_cell
from functools import reduce
import scipy.linalg as la


kmesh = [1, 1, 1]

#cell = gto.M(
#    atom = '''He     0.  0.  0. ''',
cell = gto.Cell()
# .a is a matrix for lattice vectors.





cell.a = '''
  4.03893   0.00000   0.00000
 -0.00000   4.03893   0.00000
 -0.00000  -0.00000   4.03893
'''
cell.atom = '''  
Al   0.00000   0.00000   0.00000
Al   2.01946   2.01946   0.00000
Al   2.01946   0.00000   2.01946
Al   0.00000   2.01946   2.01946
'''
#Basis: Al-ccpvDz             '''
cell.basis= {'Al': Mgto.parse('''
Al S
8.257944 0.003287
4.514245 -0.017168
2.467734 0.069766
1.348998 -0.183475
0.737436 -0.147133
0.403123 0.046882
0.220369 0.308423
0.120466 0.451564
0.065853 0.302904
0.035999 0.079545
Al S
0.236926 1.000000
Al P
1.570603 -0.002645
0.977752 -0.037850
0.608683 0.006636
0.378925 0.089291
0.235893 0.134421
0.146851 0.256105
0.091420 0.238970
0.056912 0.260677
0.035429 0.112350
0.022056 0.052665
Al P
0.202698 1.000000
Al D
0.192882 1.000000
''')
}

cell.ecp = {'Al': Mgto.basis.parse_ecp('''
Al nelec 10
Al ul
1 5.073893 3.000000
3 8.607001 15.221680
2 3.027490 -11.165685
Al S
2 7.863954 14.879513
2 2.061358 20.746863
Al P
2 3.125175 7.786227
2 1.414930 7.109015
''')
}



cell.unit='A'
cell.drop_exponent=0.1

cell.verbose = 5
cell.spin =0

cell.build()
sp_twist=[0.11,0.23,-0.34]
twist = np.asarray(sp_twist) / 1.0
kmesh=[1,1,1]
kpts = cell.make_kpts((1,1,1), with_gamma_point=False,  wrap_around=True, scaled_center=twist)

supcell=cell
mydf = df.GDF(supcell,kpts)
mydf.auxbasis = 'weigend'
mydf._cderi_to_save = 'df_ints.h5' 
mydf.build()         
mf = scf.KRHF(supcell,kpts).density_fit()



mf.with_df._cderi = 'df_ints.h5'
mf.exxdiv = 'ewald'
mf.with_df = mydf
mf.chkfile ='Al-TZ.chk'

e_scf=mf.kernel()    


ener = open('e_scf','w')
ener.write('%s\n' % (e_scf))
print('e_scf',e_scf)
ener.close()

title="Al-DZ"

from PyscfToQmcpack import savetoqmcpack
savetoqmcpack(cell,mf,title=title,kmesh=kmesh,kpts=kpts,sp_twist=kpts)



