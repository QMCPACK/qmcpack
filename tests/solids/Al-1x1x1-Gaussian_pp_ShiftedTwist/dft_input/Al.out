Discarded 24 diffused primitive functions, 0 contracted functions
#INFO: **** input file is /home/abenali/Work/Development/QMCPACK-Benali/qmcpack/tests/solids/Al-1x1x1-Gaussian_pp_ShiftedTwist/dft_input/Al.py ****
#!/usr/bin/env python
  
'''
Gamma point post-HF calculation needs only real integrals.
Methods implemented in finite-size system can be directly used here without
any modification.
'''


import numpy as np
from pyscf import lib
from pyscf.pbc import gto, scf, dft
from pyscf import gto as Mgto
from pyscf.pbc import df
from pyscf.pbc import ao2mo
from pyscf.pbc import tools
from pyscf.pbc.tools.pbc import super_cell
from functools import reduce
import scipy.linalg as la


kmesh = [1, 1, 1]

#cell = gto.M(
#    atom = '''He     0.  0.  0. ''',
cell = gto.Cell()
# .a is a matrix for lattice vectors.





cell.a = '''
  4.03893   0.00000   0.00000
 -0.00000   4.03893   0.00000
 -0.00000  -0.00000   4.03893
'''
cell.atom = '''  
Al   0.00000   0.00000   0.00000
Al   2.01946   2.01946   0.00000
Al   2.01946   0.00000   2.01946
Al   0.00000   2.01946   2.01946
'''
#Basis: Al-ccpvDz             '''
cell.basis= {'Al': Mgto.parse('''
Al S
8.257944 0.003287
4.514245 -0.017168
2.467734 0.069766
1.348998 -0.183475
0.737436 -0.147133
0.403123 0.046882
0.220369 0.308423
0.120466 0.451564
0.065853 0.302904
0.035999 0.079545
Al S
0.236926 1.000000
Al P
1.570603 -0.002645
0.977752 -0.037850
0.608683 0.006636
0.378925 0.089291
0.235893 0.134421
0.146851 0.256105
0.091420 0.238970
0.056912 0.260677
0.035429 0.112350
0.022056 0.052665
Al P
0.202698 1.000000
Al D
0.192882 1.000000
''')
}

cell.ecp = {'Al': Mgto.basis.parse_ecp('''
Al nelec 10
Al ul
1 5.073893 3.000000
3 8.607001 15.221680
2 3.027490 -11.165685
Al S
2 7.863954 14.879513
2 2.061358 20.746863
Al P
2 3.125175 7.786227
2 1.414930 7.109015
''')
}



cell.unit='A'
cell.drop_exponent=0.1

cell.verbose = 5
cell.spin =0

cell.build()
sp_twist=[0.11,0.23,-0.34]
twist = np.asarray(sp_twist) / 1.0
kmesh=[1,1,1]
kpts = cell.make_kpts((1,1,1), with_gamma_point=False,  wrap_around=True, scaled_center=twist)

supcell=cell
mydf = df.GDF(supcell,kpts)
mydf.auxbasis = 'weigend'
mydf._cderi_to_save = 'df_ints.h5' 
mydf.build()         
mf = scf.KRHF(supcell,kpts).density_fit()



mf.with_df._cderi = 'df_ints.h5'
mf.exxdiv = 'ewald'
mf.with_df = mydf
mf.chkfile ='Al-TZ.chk'

e_scf=mf.kernel()    


ener = open('e_scf','w')
ener.write('%s\n' % (e_scf))
print('e_scf',e_scf)
ener.close()

title="Al-DZ"

from PyscfToQmcpack import savetoqmcpack
savetoqmcpack(cell,mf,title=title,kmesh=kmesh,kpts=kpts,sp_twist=kpts)



#INFO: ******************** input file end ********************


System: ('Linux', 'abenali', '4.15.0-65-generic', '#74-Ubuntu SMP Tue Sep 17 17:06:04 UTC 2019', 'x86_64', 'x86_64')  Threads 16
Python 2.7.15+ (default, Oct  7 2019, 17:39:04) 
[GCC 7.4.0]
numpy 1.14.0  scipy 1.0.0
Date: Thu Oct 10 17:25:40 2019
PySCF version 1.7.0a
PySCF path  /home/abenali/Work/src/pyscf/pyscf
GIT HEAD      ref: refs/heads/dev
GIT dev branch  501bd52ce3cc3461a512f3341aac8c8be8a7369a

[CONFIG] conf_file None
[INPUT] verbose = 5
[INPUT] max_memory = 4000 
[INPUT] num. atoms = 4
[INPUT] num. electrons = 12
[INPUT] charge = 0
[INPUT] spin (= nelec alpha-beta = 2S) = 0
[INPUT] symmetry False subgroup None
[INPUT] Mole.unit = A
[INPUT]  1 Al     0.000000000000   0.000000000000   0.000000000000 AA    0.000000000000   0.000000000000   0.000000000000 Bohr
[INPUT]  2 Al     2.019460000000   2.019460000000   0.000000000000 AA    3.816226319514   3.816226319514   0.000000000000 Bohr
[INPUT]  3 Al     2.019460000000   0.000000000000   2.019460000000 AA    3.816226319514   0.000000000000   3.816226319514 Bohr
[INPUT]  4 Al     0.000000000000   2.019460000000   2.019460000000 AA    0.000000000000   3.816226319514   3.816226319514 Bohr
[INPUT] ---------------- BASIS SET ---------------- 
[INPUT] l, kappa, [nprim/nctr], expnt,             c_1 c_2 ...
[INPUT] Al
[INPUT] 0    0    [8    /1   ]  8.257944          0.003287
                                4.514245          -0.017168
                                2.467734          0.069766
                                1.348998          -0.183475
                                0.737436          -0.147133
                                0.403123          0.046882
                                0.220369          0.308423
                                0.120466          0.451564
[INPUT] 0    0    [1    /1   ]  0.236926             1
[INPUT] 1    0    [6    /1   ]  1.570603          -0.002645
                                0.977752          -0.03785
                                0.608683          0.006636
                                0.378925          0.089291
                                0.235893          0.134421
                                0.146851          0.256105
[INPUT] 1    0    [1    /1   ]  0.202698             1
[INPUT] 2    0    [1    /1   ]  0.192882             1

Ewald components = 5.17065632266529e-50, -40.3520529525604, 29.5393670563905
nuclear repulsion = -10.8126858961699
number of shells = 20
number of NR pGTOs = 140
number of NR cGTOs = 52
basis = {'Al': [[0, [8.257944, 0.003287], [4.514245, -0.017168], [2.467734, 0.069766], [1.348998, -0.183475], [0.737436, -0.147133], [0.403123, 0.046882], [0.220369, 0.308423], [0.120466, 0.451564], [0.065853, 0.302904], [0.035999, 0.079545]], [0, [0.236926, 1.0]], [1, [1.570603, -0.002645], [0.977752, -0.03785], [0.608683, 0.006636], [0.378925, 0.089291], [0.235893, 0.134421], [0.146851, 0.256105], [0.09142, 0.23897], [0.056912, 0.260677], [0.035429, 0.11235], [0.022056, 0.052665]], [1, [0.202698, 1.0]], [2, [0.192882, 1.0]]]}
ecp = {'Al': [10, [[-1, [[], [[5.073893, 3.0]], [[3.02749, -11.165685]], [[8.607001, 15.22168]]]], [0, [[], [], [[7.863954, 14.879513], [2.061358, 20.746863]], []]], [1, [[], [], [[3.125175, 7.786227], [1.41493, 7.109015]], []]]]]}
CPU time:         1.09
lattice vectors  a1 [7.632471536, 0.000000000, 0.000000000]
                 a2 [-0.000000000, 7.632471536, 0.000000000]
                 a3 [-0.000000000, -0.000000000, 7.632471536]
dimension = 3
low_dim_ft_type = None
Cell volume = 444.627
exp_to_discard = 0.1
rcut = 18.1851218647085 (nimgs = [3 3 3])
lattice sum = 251 cells
precision = 1e-08
pseudo = None
ke_cutoff = 207.1606219407866
    = [50 50 50] mesh (125000 PWs)
ew_eta = 1.98034
ew_cut = 3.4825353084680053 (nimgs = [1 1 1])


******** <class 'pyscf.pbc.df.df.GDF'> ********
mesh = [13 13 13] (2197 PWs)
auxbasis = weigend
eta = 0.2
exp_to_discard = 0.1
_cderi_to_save = df_ints.h5
len(kpts) = 1
num shells = 68, num cGTOs = 204
Drop 8 primitive fitting functions
make aux basis, num shells = 60, num cGTOs = 188
auxcell.rcut 12.314818012510043
Num uniq kpts 1


******** <class 'pyscf.pbc.scf.khf.KRHF'> ********
method = KRHF-KSCF-RHF-SCF-RHF
initial guess = minao
damping factor = 0
level shift factor = 0
DIIS = <class 'pyscf.scf.diis.CDIIS'>
DIIS start cycle = 1
DIIS space = 8
SCF tol = 1e-07
SCF gradient tol = None
max. SCF cycles = 50
direct_scf = False
chkfile to save SCF result = Al-TZ.chk
max_memory 4000 MB (current use 202 MB)


******** PBC SCF flags ********
N kpts = 1
kpts = [[ 0.09055394  0.18934006 -0.279894  ]]
Exchange divergence treatment (exxdiv) = ewald
Ewald components = 8.84516447046245e-102, -1.11818785688414, 0.93231769241698
    madelung (= occupied orbital energy shift) = 0.3717403289343113
    Total energy shift due to Ewald probe charge = -1/2 * Nelec*madelung = -2.23044197361
DF object = <pyscf.pbc.df.df.GDF object at 0x14892eca5910>
Set gradient conv threshold to 0.000316228
Big error detected in the electron number of initial guess density matrix (Ne/cell = 12.0765)!
  This can cause huge error in Fock matrix and lead to instability in SCF for low-dimensional systems.
  DM is normalized wrt the number of electrons 12.0
cond(S) = [8033.08793604]
Ewald components = 8.84516447046245e-102, -1.11818785688414, 0.93231769241698
    CPU time for vj and vk      1.44 sec, wall time      0.18 sec
Ewald components = 5.17065632266529e-50, -40.3520529525604, 29.5393670563905
E1 = (6.275129162510538-1.2531830782348727e-15j)  E_coul = (-3.2166025005775847+1.8340790188881273e-17j)
init E= -7.75415923423691
    CPU time for initialize scf    205.09 sec, wall time     34.12 sec
HOMO = 0.430408534214  LUMO = 0.477010900302
     k-point                  mo_energy
   0 ( 0.110  0.230 -0.340)   [-0.58792463 -0.14706624  0.05733521  0.21484883  0.31199655  0.43040853] [0.4770109  0.53737284 0.57936993 0.63216154 0.67105682 0.67979987
 0.71805832 0.71965896 0.84599879 0.9214259  0.92400051 0.99892982
 1.01170004 1.05290565 1.0814844  1.1507146  1.18197535 1.21319633
 1.22565857 1.25787826 1.2601097  1.32211961 1.32392687 1.38261987
 1.4465516  1.48022751 1.50234168 1.53113349 1.58858935 1.61983476
 4.05967892 5.25718968 5.40476661 5.45587266 5.51297052 5.55052065
 5.55670532 5.58142065 5.60612891 5.63185847 5.70873949 5.73138071
 6.29540334 6.5128619  6.60491681 6.66765686]
Ewald components = 8.84516447046245e-102, -1.11818785688414, 0.93231769241698
    CPU time for vj and vk      1.39 sec, wall time      0.18 sec
Ewald components = 5.17065632266529e-50, -40.3520529525604, 29.5393670563905
E1 = (5.900153475272573-6.905736269791367e-16j)  E_coul = (-3.107086189226503+9.622047119241821e-17j)
cycle= 1 E= -8.0196186101238  delta_E= -0.265  |g|= 0.208  |ddm|= 2.87
    CPU time for cycle= 1      1.92 sec, wall time      0.25 sec
HOMO = 0.143958072169  LUMO = 0.56152755392
     k-point                  mo_energy
   0 ( 0.110  0.230 -0.340)   [-0.24537884 -0.1372148  -0.03862658  0.03536718  0.07308733  0.14395807] [0.56152755 0.57763452 0.65375961 0.66710629 0.68857717 0.73222494
 0.75763588 0.75791333 0.85608832 0.92606214 0.93671173 1.00368994
 1.01274418 1.04343409 1.11280724 1.17667214 1.19633209 1.20921921
 1.24303983 1.24440256 1.25828145 1.30085108 1.30387775 1.39152548
 1.44583973 1.46030026 1.49669648 1.52781259 1.58536481 1.61854023
 4.04197187 5.23483041 5.3852569  5.43519043 5.49316038 5.53220957
 5.53552475 5.55865597 5.58704235 5.61056145 5.68553979 5.71129775
 6.27129397 6.48808414 6.58001222 6.6428722 ]
Ewald components = 8.84516447046245e-102, -1.11818785688414, 0.93231769241698
    CPU time for vj and vk      1.49 sec, wall time      0.19 sec
Ewald components = 5.17065632266529e-50, -40.3520529525604, 29.5393670563905
E1 = (5.867101031917706-6.3750400851367e-16j)  E_coul = (-3.1097233318731248+2.2656298306815117e-17j)
cycle= 2 E= -8.05530819612529  delta_E= -0.0357  |g|= 0.0941  |ddm|= 0.406
    CPU time for cycle= 2      2.17 sec, wall time      0.28 sec
HOMO = 0.138276543809  LUMO = 0.562326374631
     k-point                  mo_energy
   0 ( 0.110  0.230 -0.340)   [-0.24619233 -0.14193922 -0.04253114  0.03301851  0.06279833  0.13827654] [0.56232637 0.58624874 0.65593322 0.66647873 0.69442543 0.73247723
 0.75953782 0.7599957  0.85570245 0.92618072 0.93796246 1.00384865
 1.01362075 1.04128703 1.1146498  1.1771669  1.19599524 1.209354
 1.2424797  1.24421575 1.25962595 1.2999557  1.30357088 1.39118758
 1.44467482 1.46015823 1.49613635 1.52648255 1.58331388 1.61596143
 4.0312     5.22143285 5.3724789  5.42295296 5.47920743 5.51707092
 5.52063194 5.54666139 5.57431802 5.59701425 5.67116354 5.69717351
 6.2576411  6.4737249  6.56539076 6.62817734]
Ewald components = 8.84516447046245e-102, -1.11818785688414, 0.93231769241698
    CPU time for vj and vk      1.44 sec, wall time      0.18 sec
Ewald components = 5.17065632266529e-50, -40.3520529525604, 29.5393670563905
E1 = (5.851802982175749-6.111866419029197e-16j)  E_coul = (-3.110019525801942-1.6222613233830775e-17j)
cycle= 3 E= -8.07090243979606  delta_E= -0.0156  |g|= 0.0266  |ddm|= 0.352
    CPU time for cycle= 3      2.14 sec, wall time      0.27 sec
HOMO = 0.142244704539  LUMO = 0.562270518386
     k-point                  mo_energy
   0 ( 0.110  0.230 -0.340)   [-0.24552458 -0.13844207 -0.03954205  0.03471683  0.06954741  0.1422447 ] [0.56227052 0.58113956 0.65444457 0.66754896 0.69040151 0.73206968
 0.75803655 0.75842942 0.85561257 0.92568219 0.93690397 1.00338396
 1.0136044  1.04142183 1.11350823 1.17638352 1.19553215 1.20901723
 1.24310921 1.24393013 1.26021391 1.30083491 1.3041247  1.39114045
 1.44446026 1.46095753 1.49642353 1.52685009 1.5832136  1.61522867
 4.03112279 5.22056044 5.37139939 5.42259424 5.47875787 5.51640044
 5.51953069 5.54700772 5.57428676 5.59601316 5.67076563 5.69687912
 6.25759787 6.47367713 6.56522825 6.62775967]
Ewald components = 8.84516447046245e-102, -1.11818785688414, 0.93231769241698
    CPU time for vj and vk      0.92 sec, wall time      0.12 sec
Ewald components = 5.17065632266529e-50, -40.3520529525604, 29.5393670563905
E1 = (5.849950057907114-6.107230104783061e-16j)  E_coul = (-3.1095834078069884-7.538110157085542e-18j)
cycle= 4 E= -8.07231924606974  delta_E= -0.00142  |g|= 0.00867  |ddm|= 0.0859
    CPU time for cycle= 4      1.29 sec, wall time      0.16 sec
HOMO = 0.140167723348  LUMO = 0.562599099183
     k-point                  mo_energy
   0 ( 0.110  0.230 -0.340)   [-0.24609956 -0.14028861 -0.04105574  0.03404687  0.06560239  0.14016772] [0.5625991  0.58396787 0.65527149 0.66739188 0.69225944 0.73218567
 0.75863575 0.75900129 0.85554181 0.92588711 0.93739583 1.00360649
 1.01363784 1.04102138 1.1139135  1.17666454 1.1954325  1.20899006
 1.24300478 1.24421059 1.26062484 1.30071154 1.30409664 1.39150313
 1.44434807 1.46093943 1.49640364 1.52684481 1.58301496 1.61479227
 4.0307214  5.21965627 5.37051441 5.42209687 5.4782528  5.51571693
 5.51853523 5.54689479 5.5740593  5.59510075 5.67013635 5.6964559
 6.25700935 6.47296267 6.5644518  6.62691952]
Ewald components = 8.84516447046245e-102, -1.11818785688414, 0.93231769241698
    CPU time for vj and vk      0.99 sec, wall time      0.13 sec
Ewald components = 5.17065632266529e-50, -40.3520529525604, 29.5393670563905
E1 = (5.849489469200121-6.113975001672268e-16j)  E_coul = (-3.1092908593156348+3.9760772100411125e-17j)
cycle= 5 E= -8.07248728628538  delta_E= -0.000168  |g|= 0.00211  |ddm|= 0.0374
    CPU time for cycle= 5      1.49 sec, wall time      0.19 sec
HOMO = 0.141106786156  LUMO = 0.562557769752
     k-point                  mo_energy
   0 ( 0.110  0.230 -0.340)   [-0.24598968 -0.13962474 -0.04051553  0.03432889  0.06731677  0.14110679] [0.56255777 0.58257106 0.65488092 0.66758189 0.69146104 0.73211173
 0.75839957 0.75874883 0.85556013 0.9258248  0.93718722 1.00353956
 1.01367918 1.0411309  1.11365655 1.17649754 1.19535007 1.20893254
 1.24312061 1.2441643  1.26064466 1.30088252 1.3041808  1.39151273
 1.44431423 1.46098806 1.49638858 1.52691124 1.58303143 1.61472188
 4.03069892 5.21953701 5.37043746 5.4221133  5.47824142 5.51576575
 5.51843042 5.54691871 5.57408531 5.59495589 5.67012108 5.69648659
 6.25702903 6.47296847 6.56443969 6.62688909]
Ewald components = 8.84516447046245e-102, -1.11818785688414, 0.93231769241698
    CPU time for vj and vk      1.08 sec, wall time      0.14 sec
Ewald components = 5.17065632266529e-50, -40.3520529525604, 29.5393670563905
E1 = (5.849443499309608-6.119577727571661e-16j)  E_coul = (-3.1092520025428634-3.911565063195857e-18j)
cycle= 6 E= -8.07249439940312  delta_E= -7.11e-06  |g|= 0.000463  |ddm|= 0.00568
    CPU time for cycle= 6      1.64 sec, wall time      0.21 sec
HOMO = 0.140722806819  LUMO = 0.562569903838
     k-point                  mo_energy
   0 ( 0.110  0.230 -0.340)   [-0.24602227 -0.13986216 -0.04071409  0.03422516  0.06659622  0.14072281] [0.5625699  0.58317525 0.65505452 0.66750572 0.69177136 0.73214561
 0.75848953 0.75883393 0.85554348 0.92585814 0.9372549  1.00356708
 1.01366178 1.0410933  1.11375199 1.17655685 1.19536134 1.20894155
 1.24307614 1.24420278 1.26065582 1.30082275 1.30414815 1.39153972
 1.44430924 1.46095681 1.49637652 1.52690039 1.58302846 1.6147246
 4.03068564 5.21950788 5.37041419 5.42208384 5.47822826 5.51574693
 5.5183986  5.54688835 5.57407553 5.59492349 5.67008049 5.69647403
 6.25700143 6.47293965 6.5644092  6.62686193]
Ewald components = 8.84516447046245e-102, -1.11818785688414, 0.93231769241698
    CPU time for vj and vk      1.01 sec, wall time      0.13 sec
Ewald components = 5.17065632266529e-50, -40.3520529525604, 29.5393670563905
E1 = (5.849441466804805-6.142983100788869e-16j)  E_coul = (-3.109250326227598-9.56922899015235e-18j)
cycle= 7 E= -8.07249475559266  delta_E= -3.56e-07  |g|= 0.00017  |ddm|= 0.0011
    CPU time for cycle= 7      1.56 sec, wall time      0.20 sec
HOMO = 0.140882232049  LUMO = 0.562567424179
     k-point                  mo_energy
   0 ( 0.110  0.230 -0.340)   [-0.24601    -0.13976745 -0.0406347   0.03427056  0.06688635  0.14088223] [0.56256742 0.58292692 0.65498547 0.66753958 0.69164124 0.73213302
 0.75845328 0.75879553 0.85554938 0.92584685 0.93722861 1.0035593
 1.01367092 1.04110933 1.11371197 1.17652775 1.19535639 1.20893632
 1.24309551 1.24419028 1.26065551 1.30084782 1.30416234 1.39152605
 1.44430915 1.46097474 1.49638265 1.52690979 1.58303713 1.61472711
 4.03069404 5.21952177 5.37042957 5.4220977  5.47824606 5.51576631
 5.51841248 5.54689904 5.57409161 5.59493496 5.67009357 5.69649257
 6.25701541 6.47295668 6.56442571 6.6268791 ]
Ewald components = 8.84516447046245e-102, -1.11818785688414, 0.93231769241698
    CPU time for vj and vk      0.86 sec, wall time      0.11 sec
Ewald components = 5.17065632266529e-50, -40.3520529525604, 29.5393670563905
E1 = (5.849437162509817-6.112424666680996e-16j)  E_coul = (-3.109246090974233-3.959706974846443e-18j)
cycle= 8 E= -8.07249482463428  delta_E= -6.9e-08  |g|= 6.61e-05  |ddm|= 0.000605
    CPU time for cycle= 8      1.40 sec, wall time      0.18 sec
HOMO = 0.140846876134  LUMO = 0.562568954265
     k-point                  mo_energy
   0 ( 0.110  0.230 -0.340)   [-0.24601314 -0.13978983 -0.04065323  0.03426073  0.06682108  0.14084688] [0.56256895 0.5829812  0.65500107 0.66753261 0.69167036 0.73213603
 0.75846205 0.75880363 0.85554783 0.92584984 0.93723498 1.00356217
 1.01366968 1.04110557 1.11372064 1.17653294 1.19535782 1.20893732
 1.24309124 1.24419355 1.26065628 1.30084229 1.30415913 1.39152794
 1.44430877 1.46097191 1.49638128 1.52690894 1.5830368  1.614727
 4.03069043 5.21951697 5.37042395 5.42209174 5.4782428  5.51576183
 5.51840702 5.54689405 5.57408859 5.59493019 5.67008784 5.69648897
 6.25701011 6.47295138 6.56442037 6.62687358]
Ewald components = 8.84516447046245e-102, -1.11818785688414, 0.93231769241698
    CPU time for vj and vk      0.84 sec, wall time      0.11 sec
Ewald components = 5.17065632266529e-50, -40.3520529525604, 29.5393670563905
E1 = (5.84943721268546-6.143188241580782e-16j)  E_coul = (-3.109246147329901-4.724723310004886e-19j)
Extra cycle  E= -8.07249483081431  delta_E= -6.18e-09  |g|= 4.17e-05  |ddm|= 9.03e-05
    CPU time for scf_cycle    219.93 sec, wall time     36.01 sec
    CPU time for SCF    220.17 sec, wall time     36.07 sec
converged SCF energy = -8.07249483081431
('e_scf', -8.072494830814307)
Using Python 2.x
Wavefunction successfully saved to QMCPACK HDF5 Format
Use: "convert4qmc -pyscf  Al-DZ.h5" to generate QMCPACK input files
